src/pip/_internal/operations/prepare.py

```diff
@@ -105,23 +105,24 @@ def prep_for_dist(self, finder, build_isolation):
             # requirements.
             self.req.build_env = BuildEnvironment()
             self.req.build_env.install_requirements(
-                finder, self.req.pyproject_requires,
+                finder, self.req.pyproject_requires, 'overlay',
```

Comment:
The inverse: build requirements have priority over the backend requirements.

Reply:
Oh, are you thinking of the case where a project needs a new version of wheel, but the backend just says "wheel"? OK, I can see that. That's something that probably warrants an explicit comment in the PEP 517 implementation. I'll remember to add something. (And it may well be worth making this point explicit in the PEP itself).